Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -1,6 +1,73 @@
 #include "Training.h"
+#include <math.h>
 
-void trainingTask(uint8_t* data)
-{
+uint8_t selectBit(uint8_t bits, int index){
+	return (bits >> index) & 0x01; //Only requesting one bit, so return bit at index
+}
+
+uint8_t selectBits(uint8_t bits, int index, int end){
+	if(end < index)
+		return -1; //End must be greater than index
+	uint8_t returnVal = bits >> index;
+	uint8_t range = end - index + (index == 0 ? 1 : 0); // calculate number of bits requested
+	uint8_t maskBits = 0;
+	for(int i = 0; i < range; i++)
+		maskBits += pow(2, i);		// Determine mask bits
+	return returnVal & maskBits;
+}
```

Comment:
You should be able to use selectBits() to mask only one bit by setting the index and end to the bit you want to mask. Then you don't need to have the selectBit() function.